stl/inc/deque

```diff
@@ -1236,52 +1314,84 @@ private:
 
-        if (_UFirst != _ULast) {
-            if (_Off <= _Mysize() / 2) { // closer to front, push to front then rotate
-                _Restore_old_size_guard<_Pop_direction::_Front> _Guard{this, _Oldsize};
-                for (; _UFirst != _ULast; ++_UFirst) {
-                    emplace_front(*_UFirst); // prepend flipped
+        _Orphan_all();
+        if (_Off <= _Oldsize / 2) { // closer to front, push to front then rotate
+            _Restore_old_size_guard<_Pop_direction::_Front> _Guard{this, _Oldsize};
+            if constexpr (_Bidi == _Is_bidi::_Yes) {
+                while (_First != _Last) {
+                    _Emplace_front_internal(*--_Last); // prepend in order
                 }
-
-                _Guard._Container = nullptr;
-
-                size_type _Num = _Mysize() - _Oldsize;
-                _STD reverse(begin(), begin() + static_cast<difference_type>(_Num)); // flip new stuff in place
-                _STD rotate(begin(), begin() + static_cast<difference_type>(_Num),
-                    begin() + static_cast<difference_type>(_Num + _Off));
-            } else { // closer to back
-                _Orphan_all();
-                _Restore_old_size_guard<_Pop_direction::_Back> _Guard{this, _Oldsize};
-                for (; _UFirst != _ULast; ++_UFirst) {
-                    _Emplace_back_internal(*_UFirst);
+            } else {
+                for (; _First != _Last; ++_First) {
+                    _Emplace_front_internal(*_First); // prepend flipped
                 }
+            }
 
-                _Guard._Container = nullptr;
+            _Guard._Container = nullptr;
 
-                _STD rotate(begin() + static_cast<difference_type>(_Off),
-                    begin() + static_cast<difference_type>(_Oldsize), end());
+            const auto _Num     = static_cast<difference_type>(_Mysize() - _Oldsize);
+            const auto _Myfirst = _Unchecked_begin();
+            const auto _Mymid   = _Myfirst + _Num;
+            if constexpr (_Bidi == _Is_bidi::_No) {
+                _STD reverse(_Myfirst, _Mymid); // flip new stuff in place
             }
+            _STD rotate(_Myfirst, _Mymid, _Mymid + static_cast<difference_type>(_Off));
+            return begin() + static_cast<difference_type>(_Off);
         }
 
+        _Restore_old_size_guard<_Pop_direction::_Back> _Guard{this, _Oldsize};
+        for (; _First != _Last; ++_First) {
+            _Emplace_back_internal(*_First);
+        }
+
+        _Guard._Container = nullptr;
+
+        _STD rotate(
+            begin() + static_cast<difference_type>(_Off), begin() + static_cast<difference_type>(_Oldsize), end());
+
         return begin() + static_cast<difference_type>(_Off);
     }
 
+public:
+    template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
+    iterator insert(const_iterator _Where, _Iter _First, _Iter _Last) {
+        // insert [_First, _Last) at _Where, input iterators

```

Comment:
Do we still need the `, input iterators` comment part? We're handling bidirectionality within the helper function.

Reply:
Above, we've reworded this to say `insert input range [_First, _Last)`.

Reply:
:fire::fire::fire: